ocaml/testsuite/tests/typing-layouts-arrays/float_u_array.ml

```diff
@@ -0,0 +1,479 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 1996 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+open! Stdlib
+
+[@@@ocaml.flambda_o3]
+
+[@@@ocaml.nolabels]
+
+(* An alias for the type of arrays. *)
+type ('a : float64) t = ('a : float64) array
+
+(* Array operations *)
+
+external length : ('a : float64) array -> int = "%array_length"
```

Comment:
Why `array_length` here rather than `floatarray_length`?  (And similarly for below primitives)

Reply:
We wish to use `%array_length` and specialize that to the different layouts. This allows for using `[@layout_poly]` with it to get one version of the external that works on value/float64/int64/... all at once.